publish-extensions.js

```diff
@@ -78,19 +75,59 @@ const readFile = util.promisify(fs.readFile);
         console.log(error);
       }
 
-      // Publish the extension.
-      const options = { packagePath: path.join('/tmp/repository', extension.location || '.') };
-      if (yarn) {
-        options.yarn = true;
+      if (extension.download) {
+        if (extension.repository) {
+          console.warn('[WARN] Ignoring `repository` property because `download` was given.')
+        }
+        if (extension.checkout) {
+          console.warn('[WARN] Ignoring `checkout` property because `download` was given.')
+        }
+        if (extension.prepublish) {
+          console.warn('[WARN] Ignoring `prepublish` property because `download` was given.')
+        }
+        if (extension.location) {
+          console.warn('[WARN] Ignoring `location` property because `download` was given.')
+        }
+
+        // Download the extension package from a GitHub release
+        console.log(`Downloading ${extension.download}`);
+        await download(extension.download, '/tmp/download', { filename: 'extension.vsix' });
+
+        // Publish the extension.
+        /** @type {import('ovsx').PublishOptions} */
+        const options = { extensionFile: '/tmp/download/extension.vsix' };
+        await ovsx.publish(options);
+        console.log(`[OK] Successfully published ${id} to Open VSX!`)
+
+      } else {
+        // Clone and set up the repository.
+        await exec(`git clone --recurse-submodules ${extension.repository} /tmp/repository`);
+        if (extension.checkout) {
+            await exec(`git checkout ${extension.checkout}`, { cwd: '/tmp/repository' });
+        }
+        let yarn = await new Promise(resolve => {
+            fs.access(path.join('/tmp/repository', 'yarn.lock'), error => resolve(!error));
+        });
+        await exec(`${yarn ? 'yarn' : 'npm'} install`, { cwd: '/tmp/repository' });
+        if (extension.prepublish) {
+            await exec(extension.prepublish, { cwd: '/tmp/repository' })
+        }
+
+        // Publish the extension.
+        /** @type {import('ovsx').PublishOptions} */
+        const options = { packagePath: path.join('/tmp/repository', extension.location || '.') };
+        if (yarn) {
+            options.yarn = true;
+        }
+        await ovsx.publish(options);
+        console.log(`[OK] Successfully published ${id} to Open VSX!`)
       }
-      await ovsx.publish(options);
-      console.log(`[OK] Successfully published ${id} to Open VSX!`)
```

Comment:
Micro-nit: You could keep this single `console.log` line instead of having it at the bottom of each `if/else` branch.